esp-hal/src/lib.rs

```diff
@@ -154,6 +154,15 @@ extern "C" fn EspDefaultHandler(_interrupt: peripherals::Interrupt) {
     );
 }
 
+/// Driver initialized in blocking mode.
+pub struct Blocking;
```

Comment:
I still need to add a `Mode` trait to constrain the type params to just `Blocking` or `Async`